saleor/graphql/discount/dataloaders.py

```diff
@@ -81,6 +81,26 @@ def batch_load(self, keys):
         return [vouchers_map.get(voucher_id) for voucher_id in keys]
 
 
+class VoucherByCodeLoader(DataLoader):
+    context_key = "voucher_by_code"
+
+    def batch_load(self, keys):
+        def with_voucher_codes(voucher_codes):
+            voucher_ids = {code.voucher_id for code in voucher_codes}
+            vouchers = (
+                Voucher.objects.using(self.database_connection_name)
+                .filter(id__in=voucher_ids)
+                .all()
+            )
+            return vouchers
```

Comment:
In such case you are not ensuring that the returned vouchers are in the same order as keys.
You must ensure that in the index 1 of vouchers I will find the voucher for code provided on index 1 in keys.

Reply:
And you should also create a test with multiple vouchers that will catch such mismatch